File: app/src/main/java/bjzhou/coolapk/app/model/Result.java
package bjzhou.coolapk.app.model;

import android.text.TextUtils;

import bjzhou.coolapk.app.exceptions.ClientException;

public class Result<T> {
    private T data;
    private String message;
    private Integer status;

    public Integer getStatus() {
        return status;
    }

    public String getMessage() {
        return message;
    }

    public T getData() {
        return data;
    }

    public Result(int i, String str) {
        status = i;
        message = str;
    }

    public Result(int i, String str, T t) {
        this(i, str);
        data = t;
    }

    public boolean isSuccess() {
        return status == null || status == 1;
    }

    public int getStatusCode() {
        return status != null ? status : 1;
    }

    public ClientException checkResult() {
        if (isSuccess()) {
            return null;
        }
        if (TextUtils.isEmpty(message)) {
            message = "Empty error message";
        }
        return new ClientException(status, message);
    }
}

File: app/src/main/kotlin/bjzhou/coolapk/app/model/Result.kt
package bjzhou.coolapk.app.model

class Result<T>(val status: Int?, str: String?) {
    var data: T? = null
    var message: String = str ?: "Empty error message"
        private set

    val isSuccess: Boolean
        get() = status == null || status == 1

    val statusCode: Int
        get() = status ?: 1

}
